core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -144,260 +140,175 @@ public TFateOperation toThrift() {
-    // used to signal a TransactionRunner to stop in the case where there are too many running
-    // i.e., the property for the pool size decreased and we have excess TransactionRunners
-    private final AtomicBoolean stop = new AtomicBoolean(false);
 
-    private Optional<FateTxStore<T>> reserveFateTx() throws InterruptedException {
-      while (keepRunning.get() && !stop.get()) {
-        FateId unreservedFateId = workQueue.poll(100, MILLISECONDS);
-
-        if (unreservedFateId == null) {
-          continue;
-        }
-        var optionalopStore = store.tryReserve(unreservedFateId);
-        if (optionalopStore.isPresent()) {
-          return optionalopStore;
+      // replacement task: at this point, the existing FateExecutors that were invalidated by the
+      // config changes have started shutdown or finished shutdown. Now create any new replacement
+      // FateExecutors needed
+      for (var poolConfig : poolConfigs.entrySet()) {
+        var configFateOps = poolConfig.getKey();
+        var configPoolSize = poolConfig.getValue();
+        synchronized (fateExecutors) {
+          if (fateExecutors.stream().map(FateExecutor::getFateOps)
+              .noneMatch(fo -> fo.equals(configFateOps))) {
+            fateExecutors
+                .add(new FateExecutor<>(Fate.this, environment, configFateOps, configPoolSize));
+          }
         }
       }
 
-      return Optional.empty();
-    }
-
-    @Override
-    public void run() {
-      runningTxRunners.add(this);
-      try {
-        while (keepRunning.get() && !stop.get()) {
-          FateTxStore<T> txStore = null;
-          ExecutionState state = new ExecutionState();
-          try {
-            var optionalopStore = reserveFateTx();
-            if (optionalopStore.isPresent()) {
-              txStore = optionalopStore.orElseThrow();
-            } else {
-              continue;
-            }
-            state.status = txStore.getStatus();
-            state.op = txStore.top();
-            if (state.status == FAILED_IN_PROGRESS) {
-              processFailed(txStore, state.op);
-            } else if (state.status == SUBMITTED || state.status == IN_PROGRESS) {
+      // resize task: For each fate executor, resize the pool to match the config as necessary and
+      // submit new TransactionRunners if the pool grew, stop TransactionRunners if the pool
+      // shrunk, and potentially suggest resizing the pool if the load is consistently high.
+      synchronized (fateExecutors) {
+        for (var fateExecutor : fateExecutors) {
+          if (fateExecutor.isShutdown()) {
+            continue;
+          }
+          final var pool = fateExecutor.getTransactionExecutor();
+          final var poolName = fateExecutor.getPoolName();
+          final var runningTxRunners = fateExecutor.getRunningTxRunners();
+          final int configured = poolConfigs.get(fateExecutor.getFateOps());
+          ThreadPools.resizePool(pool, () -> configured, poolName);
```

Comment:
I have not yet explored changing this into a cached thread pool (discussed in https://github.com/apache/accumulo/pull/5263#discussion_r1917457670 thread). The changes were already pretty large and didn't want to introduce more. Writing this as a reminder to do as a potential follow on.